diff --git a/Makefile b/Makefile
index 7c47c5f..8b46465 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-KIPS := loader sm
+KIPS := loader sm pm fs_mitm creport
 
 SUBFOLDERS := libstratosphere $(KIPS)
 
diff --git a/creport/Makefile b/creport/Makefile
index 9598aea..67fd1e2 100644
--- a/creport/Makefile
+++ b/creport/Makefile
@@ -120,7 +120,7 @@ $(BUILD):
 #---------------------------------------------------------------------------------
 clean:
 	@echo clean ...
-	@rm -fr $(BUILD) $(TARGET).nso $(TARGET).elf
+	@rm -fr $(BUILD) $(TARGET).nsp $(TARGET).npdm $(TARGET).nso $(TARGET).elf
 
 
 #---------------------------------------------------------------------------------
@@ -132,7 +132,13 @@ DEPENDS	:=	$(OFILES:.o=.d)
 #---------------------------------------------------------------------------------
 # main targets
 #---------------------------------------------------------------------------------
-all	:	$(OUTPUT).nso
+all	:	$(OUTPUT).nsp
+
+ifeq ($(strip $(APP_JSON)),)
+$(OUTPUT).nsp	:	$(OUTPUT).nso
+else
+$(OUTPUT).nsp	:	$(OUTPUT).nso $(OUTPUT).npdm
+endif
 
 $(OUTPUT).nso	:	$(OUTPUT).elf
 
